collector/remote_fx.go

```diff
@@ -0,0 +1,362 @@
+// +build windows
+
+package collector
+
+import (
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+func init() {
+	registerCollector("remote_fx", NewRemoteFx, "RemoteFX Network", "RemoteFX Graphics")
+}
+
+// A RemoteFxNetworkCollector is a Prometheus collector for
+// WMI Win32_PerfRawData_Counters_RemoteFXNetwork & Win32_PerfRawData_Counters_RemoteFXGraphics metrics
+// https://wutils.com/wmi/root/cimv2/win32_perfrawdata_counters_remotefxnetwork/
+// https://wutils.com/wmi/root/cimv2/win32_perfrawdata_counters_remotefxgraphics/
+
+type RemoteFxCollector struct {
+	// net
+	BaseTCPRTT               *prometheus.Desc
+	BaseUDPRTT               *prometheus.Desc
+	CurrentTCPBandwidth      *prometheus.Desc
+	CurrentTCPRTT            *prometheus.Desc
+	CurrentUDPBandwidth      *prometheus.Desc
+	CurrentUDPRTT            *prometheus.Desc
+	TotalReceivedBytes       *prometheus.Desc
+	TotalSentBytes           *prometheus.Desc
+	UDPPacketsReceivedPersec *prometheus.Desc
+	UDPPacketsSentPersec     *prometheus.Desc
+
+	//gfx
+	AverageEncodingTime                                *prometheus.Desc
+	FrameQuality                                       *prometheus.Desc
+	FramesSkippedPerSecondInsufficientClientResources  *prometheus.Desc
+	FramesSkippedPerSecondInsufficientNetworkResources *prometheus.Desc
+	FramesSkippedPerSecondInsufficientServerResources  *prometheus.Desc
+	GraphicsCompressionratio                           *prometheus.Desc
+	InputFramesPerSecond                               *prometheus.Desc
+	OutputFramesPerSecond                              *prometheus.Desc
+	SourceFramesPerSecond                              *prometheus.Desc
+}
+
+// NewRemoteFx ...
+func NewRemoteFx() (Collector, error) {
+	const subsystem = "remote_fx"
+	return &RemoteFxCollector{
+		// net
+		BaseTCPRTT: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_base_tcp_rtt"),
+			"Base TCP round-trip time (RTT) detected in seconds",
+			[]string{"session_name"},
+			nil,
+		),
+		BaseUDPRTT: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_base_udp_rtt"),
+			"Base UDP round-trip time (RTT) detected in seconds.",
+			[]string{"session_name"},
+			nil,
+		),
+		CurrentTCPBandwidth: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "net_current_tcp_bandwidth"),
+			"TCP Bandwidth detected in thousands of bits per second (1000 bps).",
```

Comment:
What is the unit of the value here (and the corresponding udp metric)? If the value in the data matches the help text, we should probably normalize it to bytes/sec rather than base-10 kilobits/sec.

Reply:
converted to bytes/sec